hydromt_wflow/workflows/rootzoneclim.py

```diff
@@ -0,0 +1,875 @@
+        raise ValueError(
+            "Variable precip not in forcing_obs_fn"
+            )
+
+    if "pet" not in list(ds_obs.keys()):
+        raise ValueError(
+            "Variable pet not in forcing_obs_fn"
+            )
+    if ds_cc_hist != None:
+        if "precip" not in list(ds_cc_hist.keys()):
+            raise ValueError(
+                "Variable precip not in forcing_cc_hist_fn"
+                )
+    
+        if "pet" not in list(ds_cc_hist.keys()):
+            raise ValueError(
+                "Variable pet not in forcing_cc_hist_fn"
+                )
+    if ds_cc_fut != None:
+        if "precip" not in list(ds_cc_fut.keys()):
+            raise ValueError(
+                "Variable precip not in forcing_cc_fut_fn"
+                )
+    
+        if "pet" not in list(ds_cc_fut.keys()):
+            raise ValueError(
+                "Variable pet not in forcing_cc_fut_fn"
+                )
+            
+    return None
+
+def rootzoneclim(dsrun: xr.Dataset,
+                 ds_obs: xr.Dataset,
+                 ds_like: xr.Dataset, 
+                 flwdir: xr.DataArray, 
+                 ds_cc_hist: Optional[xr.Dataset]=None,
+                 ds_cc_fut: Optional[xr.Dataset]=None,
+                 return_period: list =[2,3,5,10,15,20,25,50,60,100],
+                 Imax: float =2.0, 
+                 start_hydro_year: str="Sep",
+                 start_field_capacity: str="Apr",
+                 LAI: bool=False,
+                 rooting_depth: bool=True,
+                 correct_cc_deficit: bool=False,
+                 chunksize: int=100,
+                 missing_days_threshold: int=330,
+                 logger=logger):
+    """
+    Estimates the root zone storage parameter for current observed and (optionally 
+    for) future climate-based streamflow data. 
+    The root zone storage capacity parameter is calculated per subcatchment and is 
+    converted to a gridded map at model resolution. Optionally, this function
+    can return the wflow_sbm parameter RootingDepth by dividing the root zone
+    storage parameter by (theta_s - theta_r).
+
+    The method is based on the estimation of maximum annual storage deficits based on precipitation and estimated actual evaporation time series, 
+    which in turn are estimated from observed streamflow data and long-term precipitation and potential evap. data, as explained in Bouaziz et al. (2022).
+    The main assumption is that vegetation adapts its rootzone storage capacity to overcome dry spells with a certain return period (typically 20 years for forest ecosystems). 
+    In response to a changing climtate, it is likely that vegetation also adapts its rootzone storage capacity, thereby changing model parameters for future conditions.
+    This method also allows to estimate the change in rootzone storage capacity in response to a changing climate. 
+    
+    Parameters
+    ----------
+    dsrun : xr.Dataset
```

Comment:
I see in check_inputs the variable should be named run ? We don't really have good hydromt conventions there yet which is why I opened an issue but maybe discharge (m3/s) might be more intuitive? I wonder if it's not what we use with chelsea

Reply:
Actually I had chosen the name run, based on the data conventions stated under: 
https://deltares.github.io/hydromt/latest/user_guide/data_conventions.html# (in hydrologgy section -- run for surface water runoff (overland flow + river discharge) in m3/s). Let me know if you would rather change it?

Reply:
I know... that's the Delwaq convention but it's not used anywhere else... eg sfincs and setup_riverwidth use discharge. So I think the convention may change anyway so I would say discharge m3/s (and include it in the docstrings!)

Reply:
ok! will change this!